### ISSUE TITLE
Modify training and recurrent training

### ISSUE THREAD

**Issue by tonegas** [tonegas/nnodely#66]
remove this part from the recurrent train
---------
        model_inputs = list(self.model_def['Inputs'].keys())
        state_closed_loop = [key for key, value in self.model_def['States'].items() if 'closedLoop' in value.keys()] + list(closed_loop.keys())
        state_connect = [key for key, value in self.model_def['States'].items() if 'connect' in value.keys()] + list(connect.keys())
        non_mandatory_inputs = state_closed_loop + state_connect 
        mandatory_inputs = list(set(model_inputs) - set(non_mandatory_inputs))

        n_available_samples = n_samples - prediction_samples 
        list_of_batch_indexes = list(range(n_available_samples))

        ## Remove forbidden indexes in case of a multi-file dataset
        if dataset_name in self.multifile.keys(): ## Multi-file Dataset
            if n_samples == self.run_training_params['n_samples_train']: ## Training
                start_idx, end_idx = 0, n_samples
            elif n_samples == self.run_training_params['n_samples_val']: ## Validation
                start_idx, end_idx = self.run_training_params['n_samples_train'], self.run_training_params['n_samples_train'] + n_samples
            else: ## Test
                start_idx, end_idx = self.run_training_params['n_samples_train'] + self.run_training_params['n_samples_val'], self.run_training_params['n_samples_train'] + self.run_training_params['n_samples_val'] + n_samples
            forbidden_idxs = []
            for i in self.multifile[dataset_name]:
                if i < end_idx and i > start_idx:
                    forbidden_idxs.extend(range(i-prediction_samples, i, 1))
            list_of_batch_indexes = [idx for idx in list_of_batch_indexes if idx not in forbidden_idxs]

        ## Clip the step 
        if step < 0: ## clip the step to zero
            log.warning(f"The step is negative ({step}). The step is set to zero.", stacklevel=5)
            step = 0
        if step > (len(list_of_batch_indexes)-batch_size): ## Clip the step to the maximum number of samples
            log.warning(f"The step ({step}) is greater than the number of available samples ({len(list_of_batch_indexes)-batch_size}). The step is set to the maximum number.", stacklevel=5)
            step = len(list_of_batch_indexes)-batch_size
        ## Loss vector 
        check((batch_size+step)>0, ValueError, f"The batch_size+step must be greater than 0.")
        aux_losses = torch.zeros([len(self.model_def['Minimizers']), round(len(list_of_batch_indexes)/(batch_size+step))])

        ## Update with virtual states
        self.model.update(closed_loop=closed_loop, connect=connect)
---------
This part of the code must be removed from the code executed during each epoch.
This code must be executed before the for loop of the epochs.
